pkg/vm/vm.go

```diff
@@ -1353,14 +1374,14 @@ func (v *VM) execute(ctx *Context, op opcode.Opcode, parameter []byte) (err erro
 		elems := arr.Value().([]stackitem.Item)
 		index := len(elems) - 1
 		elem := elems[index]
+		v.estack.PushItem(elem) // push item on stack firstly, to match the reference behaviour.
```

Comment:
Not related? Is it really needed?

Reply:
Related and needed. `item.Remove` may panic if trying to modify immutable item. If so, then C# will have on stack `elem`, and neo-go won't (without this change).